### DISCUSSION TITLE
Windows 10 Error - No accessories displayed

### DISCUSSION THREAD

**Discussion by joejankowski** [dgreif/ring#630] [Q&A]
Dear team:

I am having trouble configuring and getting the homebridge to display the ring sensors as accessories, I have done a little bit of troubleshooting but I am at a loss, I am not sure where the error is originating.  The issue is that the homebrigde log files show the devices, but I cannot access them either through the homebridge direct api or the accessories tab.

For example in the logs, I show this:
[5/1/2021, 8:16:11 PM] [Ring] Adding new accessory 8c26b57a-28f5-43ad-acc8-bxxxxxxxxxxx7 sensor.contact LB-Door

When I try to use the API, I get this error:
{
  "statusCode": 400,
  "message": "Service with uniqueId of '8c26b57a-28f5-43ad-acc8-bxxxxxxxxxxx7' not found.",
  "error": "Bad Request"
}

And naturally, my accessories tab is blank.

Anyone have any hints what may be causing this?  I have not tired on a second computer, since I prefer to have it working on this one.

My config file looks like this:
{
    "bridge": {
        "name": "Homebridge xxxx",
        "username": "xx:xx:xx:xx:xx:xx",
        "port": 51714,
        "pin": "xxx-xx-xxx"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "refreshToken": "eyxxxxxx...xxxxxxvA",
            "platform": "Ring",
            "hideUnsupportedServices": true
        }
    ]
}

Is there a way/format to manually add them to the config file? 

I generated the token through the Windows 10 UI in hombridge or your plugin, I kind of forget, but it was super easy.

Here are some additional data points:

Same issue on a different computer running Windows 10
It may be a Homebridge issue related to time-out, but I am not sure. I realize now that the text above says "Adding new accessory..." and that it never really completes this phase.  I have tried rolling back to older versions of home bridge, each one down to 1.2.5 and they all seem to have issues with the ring accessories.

Thanks,

joe

**Comment by dgreif** [comment]
@joejankowski can you elaborate on which "api" you are trying to use?  If you are using the homebridge plugin, then you should be using HomeKit to control/view the accessories.  Are you trying to use a HomeKit api?  If so, the id that you see in the logs won't work.  That is the _ring_ id, not the HomeKit id.

Regarding the accessories not showing up, which devices do you have?  Please note that contact sensors are somewhat buried in the little circles at the top of the room in newer versions of the Home app.

**Reply by joejankowski** [reply]
Thanks for the answer, the API that I was making the calls with was in the Homebridge UI: Plugins Tab >> Homebridge UI >> Settings >> Rest API Documentation which is at this link localhost:8581/swagger/static/index.html.

In here, I think I should be able to query each accessory by uniqueID and get the JSON response.

_/api/accessories/{uniqueId}  Get a single accessory and refresh it's characteristics._

I have to be honest, I was not using HomeKit or the Home app.  I basically have the Ring App for that.  I am trying to get the sensor status in a Windows environment so that I can develop a status panel in c#.  I was hoping that I could use the REST to query the sensor state or use a screen shot every couple of seconds and then feed that data into a C# panel with GUI LED indicators.

With that said, other then digging into your source code and the homebridge source code which would basically kill me, is there an easy way to pull the sensor state through a query or similar that is now exposed through your plugin and the homebridge combo?  

However, I still worry that I may have a configuration issue since my accessory tab is blank in Homebridge.  Am I missing something in the basic operation of how homebridge works, or is there a dependency on another plug-in or the HomeKit or Home app that is needed?

My sensors are the standard Ring 2.0 magnetic zWave sensors for windows and doors.

Thanks,

joe

**Reply by dgreif** [reply]
Rather than pulling things through homebridge, it sounds like you would be far better off using the client api, which is an npm module you can interact with directly: https://github.com/dgreif/ring#ring-client-api

**Reply by joejankowski** [reply]
@dgreif Thanks for the pointer, I am going to need a few days to dig into this.  Didn't want to be rude and leave you hanging.

**Reply by joejankowski** [reply]
@dgrief: I am struggling with your setup and config instructions section on the ring-client-api link you sent abover.  I wanted to play around with it before I posted again, but I am stuck. Unfortunately I am not as familiar with Node.JS as you guys likely are.  I am struggling to find the ring-clinet-api in the npm install folder (windows 10).  I think my npms are installed here '\User\AppData\Roaming\npm\', but I cannot find the api listed in there in the four or five module folders.  So when I try to build the setup and config example as a ts file into a js file, it is throwing an error stating it cannot find the module for the ring-client-api.  Can you please help provide any pointers for how I can get the references linked properly since I cannot find the ring-clinet-api file/folder on my local machine?

Thanks,

joe

**Reply by dgreif** [reply]
@joejankowski I created an example project that you can copy if you are still interested in setting up your own project: https://github.com/dgreif/ring-client-example
